app/src/main/java/com/example/bigowlapp/viewModel/SupervisedGroupListViewModel.java

```diff
@@ -29,6 +33,10 @@ public SupervisedGroupListViewModel() {
         return groupLiveData;
     }
 
+    public LiveData<User> getSupervisor(String supervisorUid) {
```

Comment:
You should be using a "load" method like the following:

```
    public LiveData<List<Group>> getSupervisedGroupList() {
        if (groupLiveData == null) {
            loadListOfDocumentByArrayContains();
        }
        return groupLiveData;
    }

    private void loadListOfDocumentByArrayContains() {
        groupLiveData = groupRepository.getListOfDocumentByArrayContains("supervisedUserId",
                authRepository.getCurrentUser().getUid(), Group.class);
    }
```

This is so that we don't keep pulling from the database each time we encounter the page.

Reply:
Initially, I created the "get" and "load" methods just like the example you provided. However, the difference is I'll need the "get" method to always refresh and return different supervisor of User class for different group. From your example, your "get" method will only load once, and it will always return the same "groupLiveData" afterwards. Therefore, I don't need the If statement in the "get" method. At the end of the day, I wrote the minimized "get" method and got rid of the "load".